modules/hal_nordic/nrf_802154/radio/CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+set(NRF_802154_DRIVER_ROOT ${ZEPHYR_CURRENT_MODULE_DIR}/drivers/nrf_radio_802154)
+
+include(${NRF_802154_DRIVER_ROOT}/nrf_802154_driver_sources.cmake)
+
+zephyr_library()
+
+zephyr_include_directories(${NRF_802154_DRIVER_INCLUDE_DIRS})
+
+if (CONFIG_SOC_SERIES_NRF52X)
+zephyr_library_sources(
+  ${NRF_802154_DRIVER_SOURCES_NRF52}
+  )
+elseif (CONFIG_SOC_SERIES_NRF53X)
+zephyr_library_sources(
+  ${NRF_802154_DRIVER_SOURCES_NRF53}
+  )
+else()
+    message(FATAL_ERROR "SoC unsupported by this module")
+endif()
+
+if (CONFIG_NRF_802154_SL_OPENSOURCE)
+zephyr_library_sources(
+  ${NRF_802154_DRIVER_SOURCES_DIRECT}
+  )
+
+else()
+zephyr_library_sources(
+  ${NRF_802154_DRIVER_SOURCES_SWI}
+  )
```

Comment:
This comment goes for the complete file.

I know this is existing code coming from the `hal_nordic` repo.
When CMake code is close together, creating source list or directory lists may be a sane approach.

With this PR code is moved to Zephyr, so there is an increased risk the code getting out of sync with `hal_nordic`.
As example, a user could want to use a newer `hal_nordic` with an older Zephyr or NCS release.

In such a case there could be a risk that variables are renamed between the two repos but CMake will not fail in such cases, the vars used will just be empty (undefined).
The build itself might fail, but that just makes it harder to trace back the root cause.

Or it could be a new variable introduced, as example to have SoC family specific include, like:
`NRF_802154_DRIVER_NRF53_INCLUDE_DIRS` that is not being included by a user selecting a newer revision.

Also, a minor spelling mistake like: `NRF_802145_DRIVER_INCLUDE_DIRS` will leave an empty list, but no warnings or errors in CMake.


If instead, `hal_nordic` provides a function like:
```
function(hal_nordic_sources)
  cmake_parse_arguments(HAL "DIRECT;SWI" "FAMILY;SOURCES;INCLUDES" "" ${ARGN})

  if(HAL_UNPARSED_ARGUMENTS)
    message(FATAL_ERROR "hal_nordic_sources() given unknown arguments: ${HAL_UNPARSED_ARGUMENTS}
  endif()

  if(NOT DEFINED HAL_FAMILY)
     message(FATAL_ERROR "hal_nordic_sources() missing parameter: FAMILY <NRF52|NRF53>")
  endif()

  if(DEFINED HAL_SOURCES)
    if(${HAL_FAMILY} STREQUAL NRF52)
      list(APPEND ${HAL_SOURCES} ${NRF_802154_DRIVER_SOURCES_NRF52})
    elseif(${HAL_FAMILY} STREQUAL NRF53)
      list(APPEND ${HAL_SOURCES} ${NRF_802154_DRIVER_SOURCES_NRF53})
    endif()

    if(HAL_DIRECT)
      list(APPEND ${HAL_SOURCES} ${NRF_802154_DRIVER_SOURCES_DIRECT})
    endif()

    if(HAL_SWI)
      list(APPEND ${HAL_SOURCES} ${NRF_802154_DRIVER_SOURCES_SWI})
    endif()

    set(${HAL_SOURCES} ${${HAL_SOURCES}} PARENT_SCOPE)
  endif()

  .... similar for INCLUDES .....

endfunction()
```

then Zephyr can call this function as:

```
if(CONFIG_SOC_SERIES_NRF52X)
  set(FAMILY NRF52)
elseif(CONFIG_SOC_SERIES_NRF53X)
  set(FAMILY NRF53)
endif()

if(CONFIG_NRF_802154_SL_OPENSOURCE)
  set(TYPE DIRECT)
else()
  set(TYPE SWI)
endif()


hal_nordic_sources(FAMILY ${FAMILY}
                   SOURCES  source_list
                   INCLUDES include_list
                   ${TYPE}
)

zephyr_library_sources(${source_list})

```

This gives `hal_nordic` the freedom to rename any internal variable as you might find useful.
Or in case you need to differentiate the `INCLUDES` based on soc family or something else, you may do so without require changes to Zephyr.

Introducing a new family or new options will not break anything for users not using them.

Also, if a user makes a typo, CMake will issue an error, like:
```
CMake Error at ../../modules/hal_nordic/nrf_802154/radio/CMakeLists.txt:22 (hal_nordic_sourcse):
  Unknown CMake command "hal_nordic_sourcse".
```

or if using unknown argument or unsupported family, then a proper error message is raised.